From 97a1281ad673c53d5f0c848aca50d0de4dffcbb7 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 24 Feb 2021 22:24:03 +0200
Subject: [PATCH] [SW-33611] gaudi: security: close nics macs

Recently, we have discovered an integration bug of the NICs MACs
leading to Gaudi device freeze if the MACs address space is improperly
used.
To avoid the confusion and consequences, close the MACs address space
with the RRs.

Change-Id: I27c78f6277754753dbfe00e8bac7021ce309eb5b
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 3de1ee8..6bc88d5 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1223,10 +1223,11 @@
 	LBW_RR_PCIE_DBI_COREBGN,
 	LBW_RR_PCIE_COREEND_AUX_PHY,
 	LBW_RR_PSOC0,
-	LBW_RR_PSOC1_PCIE_PMA,
-	LBW_RR_NIC0,
-	LBW_RR_NIC1,
-	LBW_RR_NIC3,
+	LBW_RR_PSOC1_NIC0_MAC_CH,
+	LBW_RR_NIC0_TS_NIC1_MAC_CH,
+	LBW_RR_NIC1_TS_NIC2_MAC_CH,
+	LBW_RR_NIC3_MAC_CH,
+	LBW_RR_NIC3_TS_NIC4_MAC_CH,
 	LBW_RR_ETR,
 };
 
@@ -1288,25 +1289,30 @@
 	max = mmPSOC_TIMESTAMP_BASE;
 	preboot_secure_lbw_rr(LBW_RR_PSOC0, min, max);
 
-	/* PSOC_GPIO0 up to NIC0_MAC_CH0: RR(LBW_RR_PSOC1_PCIE_PMA) */
+	/* PSOC_GPIO0 up to NIC0_STAT: RR(LBW_RR_PSOC1_NIC0_MAC_CH) */
 	min = mmPSOC_GPIO0_BASE - 1;
-	max = mmNIC0_MAC_CH0_BASE;
-	preboot_secure_lbw_rr(LBW_RR_PSOC1_PCIE_PMA, min, max);
+	max = mmNIC0_STAT_BASE;
+	preboot_secure_lbw_rr(LBW_RR_PSOC1_NIC0_MAC_CH, min, max);
 
-	/* NIC0_TS up to NIC1_MAC_CH0: RR(LBW_RR_NIC0) */
+	/* NIC0_TS up to NIC1_STAT: RR(LBW_RR_NIC0_TS_NIC1_MAC_CH) */
 	min = mmNIC0_TS_BASE - 1;
-	max = mmNIC1_MAC_CH0_BASE;
-	preboot_secure_lbw_rr(LBW_RR_NIC0, min, max);
+	max = mmNIC1_STAT_BASE;
+	preboot_secure_lbw_rr(LBW_RR_NIC0_TS_NIC1_MAC_CH, min, max);
 
-	/* NIC1_TS up to NIC2_MAC_CH0: RR(LBW_RR_NIC1) */
+	/* NIC1_TS up to NIC2_STAT: RR(LBW_RR_NIC1_TS_NIC2_MAC_CH) */
 	min = mmNIC1_TS_BASE - 1;
-	max = mmNIC2_MAC_CH0_BASE;
-	preboot_secure_lbw_rr(LBW_RR_NIC1, min, max);
+	max = mmNIC2_STAT_BASE;
+	preboot_secure_lbw_rr(LBW_RR_NIC1_TS_NIC2_MAC_CH, min, max);
 
-	/* NIC3_TS up to NIC4_MAC_CH0: RR(LBW_RR_NIC3) */
+	/* NIC3_MAC_CH0 up to NIC3_STAT: RR(LBW_RR_NIC3_MAC_CH) */
+	min = mmNIC3_MAC_CH0_BASE - 1;
+	max = mmNIC3_STAT_BASE;
+	preboot_secure_lbw_rr(LBW_RR_NIC3_MAC_CH, min, max);
+
+	/* NIC3_TS up to NIC4_STAT: RR(LBW_RR_NIC3_TS_NIC4_MAC_CH) */
 	min = mmNIC3_TS_BASE - 1;
-	max = mmNIC4_MAC_CH0_BASE;
-	preboot_secure_lbw_rr(LBW_RR_NIC3, min, max);
+	max = mmNIC4_STAT_BASE;
+	preboot_secure_lbw_rr(LBW_RR_NIC3_TS_NIC4_MAC_CH, min, max);
 
 	/* mmPSOC_ETR_AXICTL (single register): RR(LBW_RR_ETR) */
 	min = HL_CFG_BASE + mmPSOC_ETR_AXICTL - 1;
